src/ResourceManager/Consumption/Commands.Consumption/Cmdlets/Budget/SetAzureRmConsumptionBudget.cs

```diff
@@ -0,0 +1,276 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Consumption.Common;
+using Microsoft.Azure.Commands.Consumption.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Consumption;
+using Microsoft.Azure.Management.Consumption.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using HelpMessages = Microsoft.Azure.Commands.Consumption.Common.ParameterHelpMessages.BudgetParameterHelpMessages;
+using ParameterSetNames = Microsoft.Azure.Commands.Consumption.Common.Constants.ParameterSetNames;
+
+namespace Microsoft.Azure.Commands.Consumption.Cmdlets.Budget
+{   
+    using Budget = Management.Consumption.Models.Budget;
```

Comment:
Same comment here about InputObject and ResourceId parameter sets.

Reply:
We don't apply piping to SetBudget cmdlet, because the object from GetBudget is a model of PSBudget, and it cannot be converted to budget model as some parameters in budget model are private set. We would like to skip piping in SetBudget cmdlet.